opennms-config/src/main/java/org/opennms/netmgt/config/EventConfUtil.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed to The OpenNMS Group, Inc (TOG) under one or more
+ * contributor license agreements.  See the LICENSE.md file
+ * distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * TOG licenses this file to You under the GNU Affero General
+ * Public License Version 3 (the "License") or (at your option)
+ * any later version.  You may not use this file except in
+ * compliance with the License.  You may obtain a copy of the
+ * License at:
+ *
+ *      https://www.gnu.org/licenses/agpl-3.0.txt
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the
+ * License.
+ */
+package org.opennms.netmgt.config;
+
+import org.apache.commons.lang.StringUtils;
+import org.opennms.core.xml.JaxbUtils;
+import org.opennms.netmgt.model.EventConfEvent;
+import org.opennms.netmgt.model.EventConfSource;
+import org.opennms.netmgt.model.events.EventConfSourceMetadataDto;
+import org.opennms.netmgt.xml.eventconf.Events;
+import org.springframework.core.io.Resource;
+import org.springframework.dao.DataRetrievalFailureException;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Date;
+import java.util.concurrent.atomic.AtomicLong;
+
+public final class EventConfUtil {
+
+    private static final AtomicLong sourceIdGenerator = new AtomicLong();
+    private static final AtomicLong eventIdGenerator = new AtomicLong();
+
+    public static List<EventConfEvent> parseResourcesAsEventConfEvents(Resource configResource) throws IOException {
```

Comment:
Just annotate with @VisibleForTesting. and add a comment that this should only be used in tests.

Same thing for other methods